File: tests/human/x/hs/order_by_map.hs
module Main where

import Data.List (sortOn)

data Item = Item { a :: Int, b :: Int } deriving Show

items :: [Item]
items = [ Item 1 2, Item 1 1, Item 0 5 ]

result :: [Item]
result = sortOn (\x -> (a x, b x)) items

main :: IO ()
main = print result

File: tests/human/x/python/order_by_map.py
items = [
    {"a": 1, "b": 2},
    {"a": 1, "b": 1},
    {"a": 0, "b": 5},
]
# Sort by a then b
result = sorted(items, key=lambda x: (x['a'], x['b']))
print(result)
